docx/header.py

```diff
@@ -0,0 +1,12 @@
+class Header(object):
+    @property
+    def text(self):
+        text = ''
+        for p in self.paragraphs:
+            text += p.text + '\n'
+
+        return text
+
+    @text.setter
+    def text(self, text):
+        raise NotImplementedError('todo')
```

Comment:
Okay, I just revised this commit and merged it in: https://github.com/python-openxml/python-docx/commits/feature/header

Since it's your first one of these I'm thinking it's faster that way, we can maintain momentum and I'll just note the whys and wherefores here so you can pick it up as you go :)

You'll probably want to rebase and force push after you've digested the comments so you're building on the main branch, but as long as your commits are individually mergeable without error there's some discretion on all that. I'm going to be cherry picking them and generally there are always revisions in the process. I'll leave the rebasing decision to you.

I do recommend you study the diff and understand why each change was made because they'll come up over and over. The general patterns are remarkably consistent on this project, probably because everything is just a fancy way of reading and writing to an XML file :)

----

* All classes, functions, and methods get a full docstring at their first appearance. Follow the conventions of the rest of the library. Everyone's got a favorite; but consistency is the objective here.
* Header should inherit from ElementProxy because it will wrap the `<w:header>` element. More on that later.
* These two methods are unsupported by tests and not yet indicated by the acceptance test errors. So they come out for now. They'll get their turn in due course :)